frontend/controller/actions/group.js

```diff
@@ -407,6 +407,8 @@ export default (sbp('sbp/selectors/register', {
             chatRoomId: rootState[params.contractID].generalChatRoomId
           })
         }
+
+        sbp('okTurtles.events/emit', PENDING_TO_WELCOME)
```

Comment:
I think this needs to be done at two different locations instead of here: at the end of the `if (sendKeyRequest) {}` block and at the next else (with `console.log('@@@@@@@@ AT join[invalid state] for ' + params.contractID)`)

Reply:
>  at the end of the `if (sendKeyRequest) {}` block

I don't think that's right... This needs to happen after we've received `OP_KEY_SHARE` and have successfully sync'd the group. At the end of `if (sendKeyRequest) {}` block we have not received `OP_KEY_SHARE`